hs/Database/CourseQueries.hs

```diff
@@ -127,10 +127,15 @@ buildSession lecs tuts =
 
 -- ** Other queries
 
--- | Gets Shape, Text and Path elements for rendering graph returned as JSON
-getGraphJSON :: Int64 -> IO Response
-getGraphJSON gId =
+-- | Looks up a graph using its title then gets the Shape, Text and Path elements 
+-- for rendering graph (returned as JSON).
+getGraphJSON :: String -> IO Response
+getGraphJSON graphName =
     runSqlite databasePath $ do
+        graphEnt :: (Maybe (Entity Graph)) <- selectFirst [GraphTitle ==. graphName] []
+        gId <- case graphEnt of
```

Comment:
Don't use `-1` as a placeholder. Instead, make the `case` expression the outermost.

```haskell
graphEnt :: ...
case graphEnt of
    Nothing -> createJSONResponse ...
    Just gId ->
        sqlTexts :: ...
```